src/coreclr/jit/codegencommon.cpp

```diff
@@ -4159,8 +4159,16 @@ void CodeGen::genHomeStackSegment(unsigned                 lclNum,
     }
     emitAttr size = emitTypeSize(loadType);
 
-    int loadOffset =
-        -(isFramePointerUsed() ? genCallerSPtoFPdelta() : genCallerSPtoInitialSPdelta()) + (int)seg.GetStackOffset();
+    int loadOffset;
+    if (isFramePointerUsed())
+    {
+        loadOffset = -genCallerSPtoFPdelta();
+    }
+    else
+    {
+        loadOffset = -(int)seg.GetStackOffset() - genCallerSPtoInitialSPdelta();
+    }
+
```

Comment:
```suggestion
    int loadOffset = isFramePointerUsed() ? -genCallerSPtoFPdelta() : -genCallerSPtoInitialSPdelta();
    loadOffset += (int)seg.GetStackOffset();
```
seg.GetStackOffset() should be added to both sides, ternaries in general are good for such things. This suggestion should now fit within [coding guidelines](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md#1519-ternary-operators).

Reply:
> seg.GetStackOffset() should be added to both sides, ternaries in general are good for such things. 

Thanks!

> This suggestion should now fit within [coding guidelines](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md#1519-ternary-operators).

```
Wrong:

(a == b) ? Foo() : Bar(); // top-level ?: disallowed
```

Reply:
Acceptable:
```
x = (a == b) ? 7 : 9;
```
By "top-level ?:" they mean used as an if replacement, it's ok to use in assignments.

Reply:
I thought that the `value1` was not function's return within `condistion ? value1 : value2`.

If needed, I will update later.